torch/testing/_internal/common_utils.py

```diff
@@ -2376,27 +2376,36 @@ def noncontiguous_like(t):
         return t
 
     # Special-cases 0-dim tensors
-    if t.ndim == 0:
-        result = t.detach().unsqueeze(0).repeat_interleave(2, dim=-1)
-        if t.dtype.is_floating_point or t.dtype.is_complex:
-            result[0] = math.nan
-        else:
-            result[0] = 0
-        result.set_(result.storage(), 1, t.size(), ())
-        result.requires_grad_(t.requires_grad)
-        return result
-
-    # 1+ dim tensor case
+    zero_dim = t.ndim == 0
+    if zero_dim:
+        t = t.unsqueeze(0)
+
+    # Handle conj bit. repeat_interleave resolves the conj, which is a bit annoying.
+    is_conj = t.is_conj()
+    if is_conj:
+        t = t.conj()
     result = torch.repeat_interleave(t.detach(), 2, dim=-1)
+
+    # Choose a "weird" value that won't be accessed
     if t.dtype.is_floating_point or t.dtype.is_complex:
-        result[..., 1::2] = math.nan
+        value = math.nan
+    elif t.dtype == torch.bool:
+        value = True
     else:
-        result[..., 1::2] = 0
+        value = 12
 
-    strides = list(result.stride())
-    strides[-1] = strides[-1] * 2
-    result.set_(result.storage(), result.storage_offset(), t.size(), stride=tuple(strides))
+    if zero_dim:
+        result[0] = value
+        result.set_(result.storage(), 1, (), ())
+    else:
+        result[..., 1::2] = value
+        strides = list(result.stride())
+        strides[-1] *= 2
+        result.set_(result.storage(), result.storage_offset(), t.size(), stride=tuple(strides))
     result.requires_grad_(t.requires_grad)
+    # At the end so that the conj() is tracked by the autograd
```

Comment:
Why do we want `conj()` tracked by autograd?

Reply:
because it needs to be tracked by autograd for everything to work as expected I believe. At least, that's what's done in the OpInfos that generate conjugate inputs. I could try deactivating the tracking of the autograd in both cases if that's any better. Any thoughts @anjali411?

Reply:
I don't think it's necessary for `conj` to be tracked by autograd for these tests. We have an `OpInfo` for conj to verify it behaves correctly independently. We do want to run gradcheck with conjugated inputs, but that's about it.